jax/traceback_util.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import traceback
+import types
+
+from .api_util import wraps
+
+_jax_path = os.path.dirname(__file__)
+_include_paths = [
+    os.path.join(_jax_path, path) for path in (
+        'config.py', 'dlpack.py', 'experimental', 'lax', 'lax_linalg.py',
```

Comment:
Could we populate this list automatically somehow, or avoid the need for it? I'm sure it will rapidly become stale.

One thing that may help would be for us to finish the refactoring that moves JAX internals into a single `_internals` directory.

Reply:
I agree that this can easily become stale, and better done automatically when possible, but I'm not sure where to begin prior to a source tree restructuring. To create this list, I subjectively inspected `jax/` and picked out what seemed appropriate. I'd welcome ideas for how to avoid forming such a list, or offer that we revisit this when we restructure our source directories.

Reply:
Revisiting this comment: I ended up improving things in #4707. It's not automatic, but I believe that it's easier to maintain, especially when source files move.